### PR TITLE
HL visualization

### PR DESCRIPTION
**Pull request by akmaralAW**
Exports constraints from the final solution. 
For visualization of constraints it's Super-quick blink though, that is why increasing the interval for animation.

### CODE REVIEW

File: scripts/visualize.py [hunk 1]
```diff
@@ -186,6 +196,19 @@ def animate_func(self, i):
...
 
+      if(self.visualize_constraints):
+        if len(self.constraints["constraints"]) > k: # not all robots have constraints
+          if any(i == time * 10 for time in self.constraints["constraints"][k]["time"]):
+            index = (self.constraints["constraints"][k]["time"]).index(i / 10)
+            pos = self.constraints["constraints"][k]["states"][index]
+            xy = np.asarray(pos[:2])
+            patch_id = k + self.robot_numbers
+            self.robot_patches[patch_id][0].set(visible=True)
+            self.robot_patches[patch_id][0].center = xy
```
Comment (whoenig) [inline scripts/visualize.py:207]: Don't we sometimes have multiple constraints per timestep? At least for cases with many high-level expansions that's certainly possible.

**Review by whoenig — APPROVED** [review]
Looks nice. I think one downside is that we only visualize the constraints of the final conflict-free HL node, so we won't really see what other solutions we have discarded on the way. Another option would be to generate one picture for each expanded HL node with the conflict highlighted, and then make an animation out of it. This might look very "jumpy" as animation, though.

**Comment by akmaralAW** [comment]
Latest changes: 
1) for each expanded high-level node saves its solution and conflict detect ed among robots solution
2) when the final solution found, visualization script checks the *conflicts* folder and if it is not empty gets .png image for each node file with conflict labelled as X, and in the end saves a .mp4 file.

File: scripts/visualize.py (hunk 1 above)
Comment (akmaralAW) [reply scripts/visualize.py]: yes, with the latest/pushed changes, we don't have to care about it, thanks, nice catch!

**Pull request merged**